packages/server/lib/controllers/v1/integrations/providerConfigKey/flows/getFlows.ts

```diff
@@ -0,0 +1,87 @@
+import { asyncWrapper } from '../../../../../utils/asyncWrapper.js';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import type { GetIntegration, GetIntegrationFlows, HTTP_VERB } from '@nangohq/types';
+import type { NangoSyncConfig } from '@nangohq/shared';
+import { configService, flowService, getSyncConfigsAsStandardConfig } from '@nangohq/shared';
+import { validationParams } from '../getIntegration.js';
+
+export const getIntegrationFlows = asyncWrapper<GetIntegrationFlows>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req, { withEnv: true });
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const valParams = validationParams.safeParse(req.params);
+    if (!valParams.success) {
+        res.status(400).send({
+            error: { code: 'invalid_uri_params', errors: zodErrorToHTTP(valParams.error) }
+        });
+        return;
+    }
+
+    const { environment } = res.locals;
+    const params: GetIntegration['Params'] = valParams.data;
+
+    const integration = await configService.getProviderConfig(params.providerConfigKey, environment.id);
+    if (!integration) {
+        res.status(404).send({ error: { code: 'not_found', message: 'Integration does not exist' } });
+        return;
+    }
+
+    const availablePublicFlows = flowService.getAllAvailableFlowsAsStandardConfig();
+    const [template] = availablePublicFlows.filter((flow) => flow.providerConfigKey === integration.provider);
+    const allFlows = await getSyncConfigsAsStandardConfig(environment.id, params.providerConfigKey);
+
+    const dbFlows: NangoSyncConfig[] = [...(allFlows?.actions || []), ...(allFlows?.syncs || [])];
+    const templateFlows: NangoSyncConfig[] = [...(template?.actions || []), ...(template?.syncs || [])];
+    const finalFlows: NangoSyncConfig[] = [...dbFlows];
+
+    for (const templateFlow of templateFlows) {
+        const similar = hasSimilarFlow(templateFlow, dbFlows);
```

Comment:
What is a similar flow? Is that the right name here?

Reply:
similar in the sense a flow with the same model or endpoint, this does not mean it's the same but since it's conflicting we only allow the customer flow. I'll add some comments